src/workerd/api/global-scope.c++

```diff
@@ -520,6 +552,21 @@ void ServiceWorkerGlobalScope::sendHibernatableWebSocketError(
   KJ_IF_SOME(h, exportedHandler) {
     KJ_IF_SOME(handler, h.webSocketError) {
       auto promise = handler(js, kj::mv(websocket), js.exceptionToJs(kj::mv(e)));
+
+      // If we have a maximum event duration timeout set, we should prevent the actor from running
+      // for more than the user selected duration.
+      KJ_IF_SOME(timeout, eventTimeout) {
+        if (timeout > 0) {
+          // We must have an actor if we are delivering an hibernatable web socket message.
+          auto& actor = KJ_ASSERT_NONNULL(IoContext::current().getActor());
+
+          promise = promise.exclusiveJoin(IoContext::current().afterLimitTimeout(timeout * kj::SECONDS)
+            .then([&actor]() {
+              actor.shutdown(2, KJ_EXCEPTION(DISCONNECTED,
```

Comment:
Why not `JSG_KJ_EXCEPTION` here like the other 2?

Reply:
I edited it for testing and forgot to rollback to `JSG_KJ_EXCEPTION`